### PR TITLE
Update to streamly 8

### PR DESCRIPTION
**Pull request by jac3km4**

### CODE REVIEW

File: stack.yaml.lock [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# This file was autogenerated by Stack.
```
Comment (harendra-kumar) [inline stack.yaml.lock]: Accidentally committed this file?

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Looks good! Just the lock file should be deleted.

File: stack.yaml.lock (hunk 1 above)
Comment (jac3km4) [reply stack.yaml.lock]: wasn't sure whether to include it or not,
I've removed it now

Comment (harendra-kumar) [reply stack.yaml.lock]: Well I don't use stack much so I am not sure what the purpose of the lock file is and whether it should be committed. @psibi would know I guess.

**Comment by harendra-kumar** [comment]
Its time to update the CI configs.

**Comment by adithyaov** [comment]
@jac3km4 Thanks for the PR.
Looks like you beat me to it :) (https://github.com/psibi/streamly-bytestring/pull/12)
I'll merge it once the CI passes.

**Pull request merged**